source/api/discovery/0.1/index.md

```diff
@@ -0,0 +1,609 @@
+
+This specification describes three levels of conformance that build upon each other in terms of functionality enabled and precision of the information published. Sets of changes are published in pages, which are then collected together in a collection per publisher. Care has been take to allow the implementation of all levels to be done in as static a way as possible, rather than requiring dynamic access to a database.
+
+### 2.1. Listing Resources and their Changes
+
+There are three levels of conformance at which changes can be described. Level 0 is simply a list of the resources available.  Level 1 adds timestamps and ordering from earliest change to most recent, allowing the consumer to stop processing once it encounters a change that it has already processed. Level 2 adds in additional clarity about the types of activities, enabling the explicit creation and deletion of resources.
+
+#### 2.1.1. Level 0: Basic Resource List
+
+The core information required to provide a minimally effective set of links to IIIF resources is just the URIs of those resources. However, with the addition of a little JSON template around those URIs, we can be on the path towards a robust set of information that clients can use to optimize their processing.  
+
+Starting with the IIIF resource URIs, we add an "Update" Activity wrapper around them.  The order of the resources in the resulting list is unimportant, but each should only appear once. In terms of optimization, this approach provides no additional benefit over any other simpler list format, but is compatible with the following levels which introduce significant benefits.  This is the minimum level for interoperability, while being compatible with the more detailed patterns described below.
+
+If resources are deleted after being refered to in the resource list, the entire list should be republished without the reference to the deleted resource. Clients should also expect to encounter resource URIs that are out of date and no longer resolve to a IIIF Manifest or Collection.
+
+Example Level 0 Activity:
+
+```
+{ 
+  "type": "Update",
+  "object": {
+    "id": "https://example.org/iiif/1/manifest",
+    "type": "Manifest"  	
+  }
+}
+```
+
+#### 2.1.2. Level 1: Basic Change List
+
+The most effective information to add beyond the basic resource list is the timestamp at which the resource was last modified (including the initial modification that created it). If we know these dates, we can add them to the Activities and order the list such that the most recent activities occur last. The time at which the document update process finished is given in the `endTime` property. Consumers will then process the list of Activities in reverse order, from last to first, stopping when they encounter an Activity they have already processed in a previous run. 
```

Comment:
Suggest changes to make it clear that the `endTime` is the timestamp we are talking about:

> The most effective information to add beyond the basic resource list is the timestamp at which the resource was last modified (including the initial modification that created it). If we know these timestamps, we can add them to the Activities using the `endTime` property, the time at which the resource update process finished. Lists with multiple activities are then ordered such that the most recent activities occur last. Consumers will then process the list of Activities in reverse order, from last to first, stopping when they encounter an Activity they have already processed in a previous run.

Reply:
I also think introducing this w/ a use case might smooth things out:

When dealing with large sets of resources, it can be useful to work with only those resources that have changed since the last time the list was processed. This can be facilitated by the addition of a time stamp that indicates when a resource was last modified using the `endTime` property. Lists with multiple activities are then ordered such that the most recent activities occur last. Consumers will then process the list of Activities in reverse order, from last to first, stopping when they encounter an Activity they have already processed in a previous run.

Reply:
This has never been my forté, but why

> Lists with multiple activities are then ordered such that the most recent activities occur last [...] Consumers will then process the list of Activities in reverse order, from last to first ...

Why not reverse the order (i.e. to _most recent first_) so that the client can process the list from top to bottom rather than having to pull the whole thing into memory and reverse it? Following the 'working with large sets' use case, it seems friendlier to me to favor the client by enabling it to use a push/streaming/SAX-like parser.

Reply:
This was more explicit in the document as to why, but there was some pushback it was too much detail as to why rather than what.

The rationale is that the first pages become static resources once finished. Whereas a constantly shifting list is always dynamic.  So publishing is push() meaning that consumption is pop().

Reply:
OK I get it, and it definitely makes sense for supporting the static implementation/level 0 use case.  That said--and I'm the last to argue for hypothetical use-cases--I do think that client streaming, especially of complete change sets, isn't a stretch. Would it be too fussy to have a flag that indicates which way the list is sorted? (probably)

Reply:
:+1: to the changes. Have tried to synthesize, starting from Jon's.